### ISSUE TITLE
Returning tons of errors for the 2015-11-03 election

### ISSUE THREAD

**Issue by wmandrews** [newsdev/nyt-ap-elections#88]
```./init.sh                 
Initialize races
NOTICE:  table "races" does not exist, skipping
CREATE TABLE
ERROR:  missing data for column "initialization_data"
CONTEXT:  COPY races, line 2: ""
Initialize reporting units
NOTICE:  table "reporting_units" does not exist, skipping
CREATE TABLE
ERROR:  missing data for column "fipscode"
CONTEXT:  COPY reporting_units, line 2: ""
Initialize candidates
NOTICE:  table "candidates" does not exist, skipping
CREATE TABLE
ERROR:  missing data for column "candidateid"
CONTEXT:  COPY candidates, line 2: ""
Initialize ballot positions
NOTICE:  table "ballot_positions" does not exist, skipping
CREATE TABLE
ERROR:  missing data for column "candidateid"
CONTEXT:  COPY ballot_positions, line 2: ""
CREATE TABLE```

**Comment by eads** [comment]
Check out https://github.com/nprapps/ap-election-loader

Next up is expanding the docs to explain all this stuff.

**Comment by jeremyjbowers** [comment]
I fixed the problems with the `ap-election-loader` (see pr ) but there's still problems. Here's a sample:

```
2015-11-19 09:36:09,326 (INFO) elex (v0.0.39) : Getting candidates for election 2015-11-03
Traceback (most recent call last):
  File "/Users/jbowers/.virtualenvs/fakelections/bin/elex", line 9, in <module>
    load_entry_point('nyt-ap-elections==0.0.39', 'console_scripts', 'elex')()
  File "/Users/jbowers/.virtualenvs/fakelections/src/nyt-ap-elections/elex/cli/app.py", line 156, in main
    app.run()
  File "/Users/jbowers/.virtualenvs/fakelections/lib/python2.7/site-packages/cement/core/foundation.py", line 764, in run
    self.controller._dispatch()
  File "/Users/jbowers/.virtualenvs/fakelections/lib/python2.7/site-packages/cement/core/controller.py", line 472, in _dispatch
    return func()
  File "/Users/jbowers/.virtualenvs/fakelections/src/nyt-ap-elections/elex/cli/decorators.py", line 13, in decorated
    return fn(self)
  File "/Users/jbowers/.virtualenvs/fakelections/src/nyt-ap-elections/elex/cli/app.py", line 94, in candidates
    self.app.render(self.app.election.candidates)
  File "/Users/jbowers/.virtualenvs/fakelections/src/nyt-ap-elections/elex/parser/api.py", line 668, in candidates
    test=self.testresults
  File "/Users/jbowers/.virtualenvs/fakelections/src/nyt-ap-elections/elex/parser/api.py", line 553, in get_raw_races
    self.next_request = payload['nextrequest']
KeyError: 'nextrequest'
```

**Comment by jeremyjbowers** [comment]
So, this works fine in the console:
```
from elex.parser import api
e = api.Election(electiondate='2015-11-03', datafile=None, testresults=False, liveresults=True, is_test=False)
raw_races = e.get_raw_races()
race_objs = e.get_race_objects(raw_races)
```

Does the CLI app do some extra parsing or something?

**Comment by jeremyjbowers** [comment]
D'oh, it was my AP_API_KEY _but_ still getting errors.

```
Initialize races
NOTICE:  table "races" does not exist, skipping
CREATE TABLE
2015-11-19 10:05:14,514 (INFO) elex (v0.0.39) : Getting races for election 2015-11-03
ERROR:  invalid input syntax for type boolean: "Special General"
CONTEXT:  COPY races, line 2, column initialization_data: "Special General"
Initialize reporting units
NOTICE:  table "reporting_units" does not exist, skipping
CREATE TABLE
2015-11-19 10:05:15,902 (INFO) elex (v0.0.39) : Getting reporting units for election 2015-11-03
ERROR:  invalid input syntax for type boolean: "Jeffco Successor Direct"
CONTEXT:  COPY reporting_units, line 2, column initialization_data: "Jeffco Successor Direct"
Initialize candidates
NOTICE:  table "candidates" does not exist, skipping
CREATE TABLE
2015-11-19 10:05:21,537 (INFO) elex (v0.0.39) : Getting candidates for election 2015-11-03
ERROR:  extra data after last expected column
CONTEXT:  COPY candidates, line 2: "polid-32374,77457,1,Richard,Brown,,32374,46731"
Initialize ballot positions
NOTICE:  table "ballot_positions" does not exist, skipping
CREATE TABLE
2015-11-19 10:05:22,352 (INFO) elex (v0.0.39) : Getting ballot positions for election 2015-11-03
ERROR:  extra data after last expected column
CONTEXT:  COPY ballot_positions, line 2: "polid-2,12482,1,Jeffco School Dist 1,Yes,2,6212,Recall Williams Dist 1"
CREATE TABLE
```
I think there's a field-table mismatch? This does feel v. brittle.

**Comment by jeremyjbowers** [comment]
@eads We should talk about this -- I think the problem isn't with the CLI but with the bash files that create / update tables.

**Comment by eads** [comment]
I don't think it's brittle so much as immature. I think more development of the reference loaders and some schema helpers in the app will help a lot.

**Comment by eads** [comment]
Yeah, that's exactly what I meant in my reply. One thing that would greatly
reduce the brittleness is something which spits out a schema.

Then you could do something like:

```
psql -c `elex races-schema psql`
# ... loading code ...
```

Or something. I'm not sure what the interface should look like, but it
seems this would really help.

**Comment by jeremyjbowers** [comment]
Yeah, I'll make a ticket for a schema generator that matches the output of the CSVs. And I'll fix this thing, too.

**Comment by jeremyjbowers** [comment]
Fixed in sha e6992c54cdfe75b6fcd6a262523340a338250ffa and https://github.com/nprapps/ap-election-loader/pull/2.
